styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java

```diff
@@ -152,19 +163,23 @@ private void tryTriggering(Workflow workflow,
   private void tryTriggering0(Workflow workflow,
                                  TriggerInstantSpec instantSpec,
                                  Set<WorkflowId> enabledWorkflows) {
-    // Ignore flyte workflows
-    if (enabledWorkflows.contains(workflow.id()) && workflow.configuration().flyteExecConf().isEmpty()) {
+
+    boolean isFlyteWorkflow = workflow.configuration().flyteExecConf().isPresent();
+
+    if (isFlyteWorkflow) {
+      log.info("Skip triggering flyte workflow");
+    } else if (enabledWorkflows.contains(workflow.id())) {
```

Comment:
It's better here to do
```
    if (isFlyteWorkflow) {
      log.info("Skip triggering flyte workflow");
      return;
    }

    if (enabledWorkflows.contains(workflow.id())) {...
```

this way we separate from the other (unrelated logic), and we return early and safe

Reply:
I did it this way since I want this code to execute: https://github.com/spotify/styx/pull/830/commits/5fbba776fb6f5edc968d52fecc7d8001ec8edf44#diff-8681f2f92e0c50ca9a99b3fc6ee3e4a2R196 So that the next trigger date updates

Reply:
Do we want this to happen? and run the `updateNextNaturalTrigger` regardless if it will fail next time?

Reply:
I think so. I assume we want to handle it in the same way as disabled workflows. Otherwise, I think that once we enable the ability to run flyte workflows that each workflow would backfill from the date it was created until we enable the functionality. Not that this would maybe be a huge problem, but seems better to not cause them to backfill.

Reply:
This is my understanding of what happens. But would be great with a second pair of eyes.